pkg/pillar/cmd/vaultmgr/vaultmgr.go

```diff
@@ -1083,25 +1114,51 @@ func handleVaultKeyFromControllerImpl(ctxArg interface{}, key string,
 		//Log the type of key used for unlocking default vault
 		log.Noticef("%s unlocked using key type %s", defaultVault,
 			etpm.CompareLegacyandSealedKey().String())
-
-		//Mark the default vault as unlocked
-		ctx.defaultVaultUnlocked = true
-
-		//publish vault key to Controller
-		if err := publishVaultKey(ctx, types.DefaultVaultName); err != nil {
-			log.Errorf("Failed to publish Vault Key, %v", err)
+	} else {
+		// We are here if we receive no keys from controller
+		if ctx.defaultVaultUnlocked {
+			// if already unlocked, do nothing
+			return
+		}
+		if !vault.IsVaultCleanupAllowed() {
+			log.Warnf("Vault cleanup is not allowed")
+			return
```

Comment:
Shouldn't we still publish vault status in this case? Or is the assumption that it couldn't possibly have changed since we received no key from zedagent/controller?

Reply:
If we do not allow to clean and recreate vault, this branch is noop, as we do nothing, i.e. no update, no new information to publish.